quarkus-restlight-commons/src/main/java/io/esastack/quarkus/restlight/commons/ReflectionInfoUtil.java

```diff
@@ -0,0 +1,107 @@
+package io.esastack.quarkus.restlight.commons;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+
+public final class ReflectionInfoUtil {
+    private static final String yyyyMMddHHmmss = "yyyy-MM-dd HH:mm:ss";
+    private static ObjectMapper DEFAULT_OBJECT_MAPPER = initDefaultMapper();
+    private static final JavaType REFLECTION_CONFIG_TYPE = DEFAULT_OBJECT_MAPPER
+            .getTypeFactory()
+            .constructType(new TypeReference<List<ReflectedClassInfo>>() {
+            }.getType());
+
+    private ReflectionInfoUtil() {
+    }
+
+    public static List<ReflectedClassInfo> loadReflections(String reflectionsFrom,
+                                                           Class<?> classInJar) throws IOException {
+        List<ReflectedClassInfo> reflectedInfos = loadReflectionsByJson(classInJar,
+                "META-INF/native-image/io.esastack/" + reflectionsFrom + "/reflection-config.json");
+        reflectedInfos.addAll(loadReflectionsBySpis(classInJar, SpiUtil.getAllSpiPaths(classInJar)));
+
+        return reflectedInfos;
+    }
+
+    private static List<ReflectedClassInfo> loadReflectionsBySpis(Class<?> classInJar, List<String> spiPaths)
+            throws IOException {
+        checkNotNull(classInJar, "classInJar");
+        checkNotNull(spiPaths, "spiPaths");
+
+        String jarPath = classInJar.getProtectionDomain().getCodeSource().getLocation().getPath();
+        final JarFile jar = new JarFile(jarPath);
+        List<ReflectedClassInfo> reflectionConfig = new ArrayList<>(12);
+
+        for (String spiPath : spiPaths) {
+            ZipEntry entry = jar.getEntry(spiPath);
+            if (entry == null) {
+                throw new IllegalStateException(spiPath + " is not exist in jar(" + jarPath + ")!");
+            }
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(jar.getInputStream(entry)));
+            String className = reader.readLine();
+            while (className != null) {
+                ReflectedClassInfo classInfo = new ReflectedClassInfo(className, null);
+                reflectionConfig.add(classInfo);
+                className = reader.readLine();
+            }
+            reader.close();
```

Comment:
Please make sure the `reader` should be closed even if the `exception` is thrown at `while() {}`.